src/expand-uri-template-with-parameters.coffee

```diff
@@ -30,32 +31,38 @@ expandUriTemplateWithParameters = (uriTemplate, parameters) ->
     for uriParameter in uriParameters
       if Object.keys(parameters).indexOf(uriParameter) is -1
         ambiguous = true
-        text = "\nAmbiguous URI parameter in template: #{uriTemplate} " + \
-               "\nParameter not defined in API description document: " + \
-               "'" + uriParameter + "'"
-        result.warnings.push text
+        result.warnings.push("""\
+          Ambiguous URI parameter in template: #{uriTemplate}
+          Parameter not defined in API description document: #{uriParameter}\
+        """)
 
-    if ambiguous is false
+    unless ambiguous
       toExpand = {}
       for uriParameter in uriParameters
         param = parameters[uriParameter]
-        if param.required is true
-          if not param.example
-            ambiguous = true
-            text = "\nAmbiguous URI parameter in template: #{uriTemplate} " + \
-                   "\nNo example value for required parameter in API description document: " + \
-                   "'" + uriParameter + "'"
-            result.warnings.push text
-          else
-            toExpand[uriParameter] = param.example
+
+        if param.example
+          toExpand[uriParameter] = param.example
+        else if param.default
+          toExpand[uriParameter] = param.default
         else
-          if param.example
-            toExpand[uriParameter] = param.example
-          else if param.default
-            toExpand[uriParameter] = param.default
+          if param.required
+            ambiguous = true
+            result.warnings.push("""\
+              Ambiguous URI parameter in template: #{uriTemplate}
+              No example value for required parameter in API description \
+              document: #{uriParameter}\
+            """)
+
+        if param.required and param.default
+          result.warnings.push("""\
+            Required URI parameter has a default value: #{uriParameter}
```

Comment:
Sorry for a nitpick. I would reword it like this: `Required URI parameter "#{uriParameter}" has a default value.`